### PR TITLE
fix #2154 conda remove should complain if requested removals don't exist

### LINKED ISSUE #2154: conda remove doesn't complain if only some of the packages don't exist
If I do `conda remove foo bar` and neither of them are in the current environment then I get an error.  However if I do `conda remove foo bar` and `foo` is present but `bar` is not then `foo` gets removed but there is no error or warning that `bar` wasn't there and was not removed.  I discovered this because I misunderstood the behavior of the `--features` flag, so it also applies to removing features.

I would suggest that a warning is required as a minimum, and perhaps it should be an error to be "symmetric" with a `conda install foo bar` where `foo` exists to be installed but `bar` does not (which just reports an error, and does not install anything).

### PR DESCRIPTION
**Pull request by marcelotrevisani**
fix #2154 
Bug fix about the Issue 2154. With this implementation Conda raises
an exception when the user tries to remove a non-existent package.

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
@kalefranz , Could you please review that? And it's kinda weird why this is falling in parallel tests, the first time I ran that failed a different test. It seems the parallel tests are kinda a flake (some of them)

**Comment by kalefranz** [comment]
@marcelotrevisani I've added a commit to this PR that I think gets us closer.  What do you think?

File: conda/core/solve.py [hunk 1]
```diff
@@ -195,7 +195,8 @@ def solve_final_state(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
                 # package with a feature that matches the track_feature. The
                 # `graph.remove_spec()` method handles that for us.
                 log.trace("using PrefixGraph to remove records for %s", spec)
-                removed_records.extend(graph.remove_spec(spec))
+                spec_remove = graph.remove_spec(spec)
```
Comment (kalefranz) [inline conda/core/solve.py]: This line is no longer needed.

File: tests/test_create.py [hunk 2]
```diff
@@ -995,14 +1004,6 @@ def test_update_deps_flag_present(self):
...
             assert package_is_installed(prefix, 'flask')
 
-    # @pytest.mark.skipif(not on_win, reason="shortcuts only relevant on Windows")
-    # def test_shortcut_in_underscore_env_shows_message(self):
-    #     prefix = make_temp_prefix("_" + str(uuid4())[:7])
-    #     with make_temp_env(prefix=prefix):
-    #         stdout, stderr = run_command(Commands.INSTALL, prefix, "console_shortcut")
-    #         assert ("Environment name starts with underscore '_'.  "
-    #                 "Skipping menu installation." in stderr)
-
```
Comment (kalefranz) [inline tests/test_create.py]: Let's put this one back.  The rest of the commented out test code is fine to be gone.

**Comment by marcelotrevisani** [comment]
Sure! Looks good to me!
:)

File: tests/test_create.py (hunk 2 above)
Comment (marcelotrevisani) [reply tests/test_create.py]: Sure. But I just want to know, why these commented code are necessary?

**Comment by kalefranz** [comment]
Looks like I created new problems :-/

**Comment by marcelotrevisani** [comment]
hahaha, sometimes this also happens. Do you want to fix that? I think there is some error in your commit because the test is failing. Just let me know if you need help! :)

**Comment by kalefranz** [comment]
Should be good to go.  I'll double-check that all tests pass in master after merge.

**Pull request merged**